crates/sqlexec/src/extension_codec.rs

```diff
@@ -357,6 +358,36 @@ impl<'a> PhysicalExtensionCodec for GlareDBExtensionCodec<'a> {
                     limit,
                 })
             }
+            proto::ExecutionPlanExtensionType::RemoteInsertExec(ext) => {
+                let provider_id = Uuid::from_slice(&ext.provider_id).map_err(|e| {
+                    DataFusionError::Plan(format!("failed to decode provider id: {e}"))
+                })?;
+                let input = ext.input.ok_or_else(|| {
+                    DataFusionError::Plan("input is required to insert data".to_string())
+                })?;
+                let schema = input
+                    .schema
+                    .ok_or_else(|| DataFusionError::Plan("schema is required".to_string()))?;
+                let input = ClientExchangeRecvExec {
+                    broadcast_id: Uuid::from_slice(&input.broadcast_id).map_err(|e| {
+                        DataFusionError::Plan(format!("failed to decode broadcast id: {e}"))
+                    })?,
+                    schema: Arc::new((&schema).try_into()?),
+                };
+
+                let provider = self
+                    .table_providers
+                    .expect("remote context should have provider cache")
+                    .get(&provider_id)
+                    .ok_or_else(|| {
+                        DataFusionError::Internal(format!("Missing proivder for id: {provider_id}"))
+                    })?;
+
+                Arc::new(RemoteInsertExec {
+                    provider: ProviderReference::Provider(provider),
+                    input: Arc::new(input),
```

Comment:
We'll want to do `input: inputs[0].clone()` here instead of trying to deserialize it.